examples/triangles.txt

```diff
@@ -0,0 +1,155 @@
+#version 300 es
+#define S(a, b, t) smoothstep(a, b, t)
+
+// Shadertoy: https://www.shadertoy.com/view/wdsfRl
+// Fork of Plexus with triangles by latitude: https://www.shadertoy.com/view/wtcXWr
+
+precision highp float;
+
+out vec4 glFragColor;
+in vec2 fragCoord;
+uniform vec3 iResolution;
+uniform float iGlobalTime;
+uniform float iTime;
+vec2 iMouse;
+
+float distLine(vec2 p, vec2 a, vec2 b){
+	vec2 pa = p - a;
+    vec2 ba = b - a;
+    float t = clamp(dot(pa, ba) / dot(ba, ba), 0.0, 1.0);
+    return length(pa - ba*t);
+}
+
+float line(vec2 p, vec2 a, vec2 b){
+	float d = distLine(p, a, b);
+    float m = S(0.03, 0.01, d);
+    float d2 =  length(a - b);
+    m *= S(1.2, 0.8, d2) * 0.5 + S(0.05, 0.03, abs(d2 - 0.75));
+    return m;
+}
+
+float distTriangle(in vec2 p, in vec2 p0, in vec2 p1, in vec2 p2 )
+{
+	vec2 e0 = p1 - p0;
+	vec2 e1 = p2 - p1;
+	vec2 e2 = p0 - p2;
+
+	vec2 v0 = p - p0;
+	vec2 v1 = p - p1;
+	vec2 v2 = p - p2;
+
+	vec2 pq0 = v0 - e0*clamp( dot(v0,e0)/dot(e0,e0), 0.0, 1.0 );
+	vec2 pq1 = v1 - e1*clamp( dot(v1,e1)/dot(e1,e1), 0.0, 1.0 );
+	vec2 pq2 = v2 - e2*clamp( dot(v2,e2)/dot(e2,e2), 0.0, 1.0 );
+    
+    float s = sign( e0.x*e2.y - e0.y*e2.x );
+    vec2 d = min( min( vec2( dot( pq0, pq0 ), s*(v0.x*e0.y-v0.y*e0.x) ),
+                       vec2( dot( pq1, pq1 ), s*(v1.x*e1.y-v1.y*e1.x) )),
+                       vec2( dot( pq2, pq2 ), s*(v2.x*e2.y-v2.y*e2.x) ));
+
+	return -sqrt(d.x)*sign(d.y);
+}
+
+float triangle(vec2 p, vec2 a, vec2 b, vec2 c){
+	float d = distTriangle(p, a, b, c);
+    float m = S(0.03, 0.01, d);
+    float d2 =  length(a - b);
+    m *= S(1.2, 0.8, d2) * 0.5 + S(0.05, 0.03, abs(d2 - 0.75));
+    return m;
+}
+
+float N21(vec2 p){
+	p = fract(p * vec2(233.34, 851.73));
+    p += dot(p, p + 23.45);
+    return fract(p.x * p.y);
```

Comment:
This is a really nice example. Again, only a minor point, but if you could use either tabs or spaces here for the indentation, that would be appreciated.